### PR TITLE
[TEST] home page stats #9384

### LINKED ISSUE #9384: [OTHER] <Simple data loads test for home page> /tests/home.spec.js
### What would you like to share?

From the home page we want to test that the data loads.

home_data_load

Specifically, we want to compare the abbreviated number generated by the abbreviateNumber function with database data as parameter to the number generated on the rendered home page. 

### Do you want to work on this issue?

Yes

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by kbventures**
## Fixes Issue
CLOSES [OTHER] <Simple data loads test for home page> /tests/home.spec.js #9384

closes #9384

## Changes proposed

From the home page we want to test that the data loads.

Specifically, we want to compare the abbreviated number generated by the abbreviateNumber function with database data as parameter to the number generated on the rendered home page.

## Check List (Check all the applicable boxes) 

- [x ] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] All new and existing tests passed.
- [ x] This PR does not contain plagiarized content.
- [ x] The title of my pull request is a short description of the requested changes.

## Screenshots

image

### CODE REVIEW

**Comment by eddiejaoude** [comment]
I tested this with a new/empty database and it fails like on GitHub Actions, but if I use an existing database it works fine - so I think this PR needs to popular the DB with some stats. Let me know if you need any help with that, but I think the data can be inserted in the test setup file https://github.com/EddieHubCommunity/BioDrop/blob/main/tests/setup/global-setup.js

**Comment by kbventures** [comment]
> `Run npx playwright test --shard=2/5
The client environment variables are valid!
The server environment variables are valid!
ReferenceError: e is not defined

   at setup/global-setup.js:9[4](https://github.com/EddieHubCommunity/BioDrop/actions/runs/6551509530/job/17792926030?pr=9462#step:7:5)

  92 |
  93 |   if (githubData.error) {
> 94 |     return logger.error(e, error);
     |                         ^
  9[5](https://github.com/EddieHubCommunity/BioDrop/actions/runs/6551509530/job/17792926030?pr=9462#step:7:6) |   }
  9[6](https://github.com/EddieHubCommunity/BioDrop/actions/runs/6551509530/job/17792926030?pr=9462#step:7:7) |
  9[7](https://github.com/EddieHubCommunity/BioDrop/actions/runs/6551509530/job/17792926030?pr=9462#step:7:8) |   await Profile.findOneAndUpdate(

    at module.exports (/home/runner/work/BioDrop/BioDrop/tests/setup/global-setup.js:94:25)

[1[8](https://github.com/EddieHubCommunity/BioDrop/actions/runs/6551509530/job/17792926030?pr=9462#step:7:9):50:35.058] INFO (2240): DB connection successful:
[[18](https://github.com/EddieHubCommunity/BioDrop/actions/runs/6551509530/job/17792926030?pr=9462#step:7:19):50:35.185] INFO (2240): github info fetched for repo: EddieHubCommunity/BioDrop
[18:50:35.185] ERROR ([22](https://github.com/EddieHubCommunity/BioDrop/actions/runs/6551509530/job/17792926030?pr=9462#step:7:23)40):
    message: "API rate limit exceeded for 40.83.43.50. (But here's the good news: Authenticated requests get a higher rate limit. Check out the documentation for more details.)"
    documentation_url: "https://docs.github.com/rest/overview/resources-in-the-rest-api#rate-limiting"
Error: Process completed with exit code 1.`

The error is unrelated to this issue.

**Comment by eddiejaoude** [comment]
> The error triggered in the CIDC is unrelated to this issue.

I think the error has never been thrown before, maybe that is why it is failing now, I will take a closer look

File: tests/home.spec.js [hunk 1]
```diff
@@ -49,3 +51,23 @@ test.describe("accessibility tests (dark)", () => {
...
 });
+
+test.describe("Data rendered correctly on home page", () => {
+  test("Total active rendered", async ({ page }) => {
+    const { stats: totalStats } = await getTotalStats();
+    const abbreviatedActive = abbreviateNumber(totalStats.active);
+    await page.goto("/");
+    const activeLocator = await page.locator("dl > div:nth-child(1) dd div:nth-child(1)");
+    const finalTotalActive = await activeLocator.textContent();
+    expect(finalTotalActive).toEqual(abbreviatedActive);
```
Comment (eddiejaoude) [inline tests/home.spec.js:62]: I think using the card text might be a better selector

```suggestion
    const section = page.getByText("Active Users");
    const parent = section.locator("..");
    const item = parent.locator("dd div:nth-child(1)");
    await expect(item).toHaveText(abbreviatedActive);
```

File: tests/home.spec.js [hunk 2]
```diff
@@ -49,3 +51,24 @@ test.describe("accessibility tests (dark)", () => {
...
+    await expect(item).toHaveText(abbreviatedActive);
+  });
+
+  test("Profile views rendered", async ({ page }) => {
+    const { stats: totalStats } = await getTotalStats();
+    const abbreviatedViews = abbreviateNumber(totalStats.views);
+    await page.goto("/");
+    const viewsLocator = await page.locator("dl > div:nth-child(2) dd div:nth-child(2)");
+    const finalTotalViews = await viewsLocator.textContent();
+    expect(finalTotalViews).toEqual(abbreviatedViews);
```
Comment (eddiejaoude) [inline tests/home.spec.js:72]: Same here for the selectors

```suggestion
    const section = page.getByText("Profile Views");
    const parent = section.locator("..");
    const item = parent.locator("dd div:nth-child(1)");
    await expect(item).toHaveText(abbreviatedViews);
```

**Review by eddiejaoude — APPROVED** [review]
Looks great, thank you 👍

**Pull request merged**